torch/_inductor/codegen/cpp_prefix.h

```diff
@@ -60,60 +61,148 @@ struct IsVecType<at::vec::Vectorized<T>>: std::true_type {};
 
 template <typename T>
 struct WeightRecp {
-  using scalar_t = typename T::value_type;
-  int64_t N;
-  std::vector<scalar_t> weight_recps;
-  WeightRecp(int64_t N) : N(N) {
-    weight_recps.reserve(N);
-    for (const auto i : c10::irange(N)) {
-      weight_recps.push_back(
-          scalar_t(static_cast<double>(1) / static_cast<double>(i + 1)));
+  int64_t outer_size;
+  int64_t main_size;
+  int64_t tail_size;
+  std::vector<T> weight_recps;
```

Comment:
Why we change the type from `T::value_type` to `T`?

Reply:
We use T , i.e., vec type instead of scalar type, as all element values of weight(vec type) may no longer be identical in masked vec welford reduce since weight may be masked.